docker/Dockerfile

```diff
@@ -8,134 +8,35 @@
 # if it isn't done, bad things will happen to local builds and CI using mob tool
 # See mob tool comments for extended discussion
 
-# builder-install contains dependencies but doesn't copy our sources in, so it can be
-# cached in google cloud.
-#
+# builder-install contains dependencies but doesn't copy
+# our sources in, so it can be cached in google cloud.
+
+
 # ##################################### #
 #  builder-install - Build environment  #
 # ##################################### #
-
 # Ubuntu 18.04
 FROM ubuntu@sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02ecb6e8 AS builder-install
 
 # Party like it's June 8th, 1989.
 #
 SHELL ["/bin/bash", "-c"]
 
-# Certain Installers make 'installations' easier by having a nice front-end. While this is great when you have a manual install, this becomes an issue during automated installations.
-ENV DEBIAN_FRONTEND=noninteractive
+# Specify the rust-toolchain for init script.
+COPY rust-toolchain /tmp
+ENV RUST_TOOLCHAIN_PATH=/tmp/rust-toolchain
 
-# Install build tools and dependencies
-#
-RUN apt-get update -q -q && \
- apt-get upgrade --yes --force-yes && \
- apt-get install --yes --force-yes \
-  alien \
-  autoconf \
-  automake \
-  binutils-dev \
-  build-essential \
-  clang \
-  cmake \
-  curl \
-  git \
-  jq \
-  libclang-dev \
-  libcurl4-openssl-dev \
-  libdw-dev \
-  libiberty-dev \
-  libpq-dev \
-  libprotobuf-c-dev \
-  libprotobuf-dev \
-  libssl-dev \
-  libssl1.1 \
-  libsystemd-dev \
-  libtool \
-  libxml2-dev \
-  llvm-dev \
-  nano \
-  nginx \
-  ninja-build \
-  ocaml-native-compilers \
-  ocamlbuild \
-  patch \
-  pkg-config \
-  postgresql-10 \
-  prometheus \
-  protobuf-c-compiler \
-  protobuf-compiler \
-  psmisc \
-  python \
-  python3-pip \
-  systemd \
-  unzip \
-  uuid-dev \
-  wget \
-  zlib1g-dev
-# psmisc = killall
-# prometheus = helps with running slam scripts locally
+COPY init_debian.sh /tmp
+RUN bash /tmp/init_debian.sh
 
-# Install go 1.16 release
-RUN wget https://golang.org/dl/go1.16.4.linux-amd64.tar.gz && \
-    mv go1.16.4.linux-amd64.tar.gz go.tgz && \
-    tar -C /usr/local -xzf go.tgz  && \
-    rm -rf go.tgz
-ENV PATH="/usr/local/go/bin:$PATH"
+# Unfortunately we cannot update ENV from the init script.
+# See https://github.com/docker/docker/issues/29110
 ENV GOPATH=/opt/go/
-ENV PATH=$PATH:$GOPATH/bin
```

Comment:
Please test that building this and go grpc gateway in the `./mob prompt` still works after this change. We should probably keep the docker ENV directive.

Reply:
I guess CI will test this as long as the new image is pushed